src/bika/lims/browser/fields/interimfieldsfield.py

```diff
@@ -74,25 +76,52 @@ class InterimFieldsField(RecordsField):
     })
     security = ClassSecurityInfo()
 
-    security.declarePrivate("get")
-
     def get(self, instance, **kwargs):
-        an_interims = RecordsField.get(self, instance, **kwargs) or []
-        if not IAnalysisService.providedBy(instance):
-            return an_interims
+        # local set interims
+        interims = RecordsField.get(self, instance, **kwargs) or []
+
+        # make sure we have a copy of the interims field
+        interims = copy.deepcopy(interims)
+
+        # return "additional result values" for analysis service
+        if IAnalysisService.providedBy(instance):
+            return interims
+
+        # return calculation interims
+        if ICalculation.providedBy(instance):
+            return interims
 
-        # This instance implements IAnalysisService
+        # retain local interims for routine analysis
+        if IRoutineAnalysis.providedBy(instance) and interims:
+            return interims
```

Comment:
When we remove this and the calculation changes, all existing analyses will get the new interims.

In my opinion we should completely remove the logic from the interims field getter and provide explicit getters for Calculation, Service and Analysis.

However, I was not sure about the expected behavior for QC analyses. Can you support here?

Reply:
> When we remove this and the calculation changes, all existing analyses will get the new interims.

So, we assume here that if an analysis has interim fields set, those coming from calculation are already in there, right?

> In my opinion we should completely remove the logic from the interims field getter and provide explicit getters for Calculation, Service and Analysis.

You're probably right. It will become less confusing

> However, I was not sure about the expected behavior for QC analyses. Can you support here?

QC Analyses should work exactly the same way as RoutineAnalyses. The only difference is that for QC analyses, user should be able to choose instruments that are not valid due to a previous QC failure. Note that out-of-date instruments (invalid certificate) should not be available to any analyses, regardless of their type.